packages/dapp/src/components/HighlightModal/Footer.js

```diff
@@ -13,18 +15,43 @@ const greenButtonDisabledStyle = `w-32 px-4 py-2 bg-gray-600 shadow shadow-green
 
 const Footer = () => {
   const { state, dispatch } = useContext(HighlightContext);
-  const [{ data, error }, connect] = useConnect()
+  const {
+    activeConnector,
+    connect,
+    connectors,
+    error,
+  } = useConnect();
+  const { data: signer } = useSigner();
+  const signatureNFTContract = useContract({
+    addressOrName: addresses(1337).signatureNFT, // @todo - need a better way to manage this, the chainid from provider will be a promise, and probably can't be used here
+    contractInterface: abis.signatureNFT,
+    signerOrProvider: signer,
+  })
+
   const isImage = !!state.image;
 
   const mint = async () => {
```

Comment:
I have a little concern with this function, that it executes arweave upload every time a user clicks `mint` but the user then might or might not actually mint their NFT - they might just reject the transaction request on their wallet

open to suggestions here, on how to bundle these two executions so we don't end up being spammed, or just paying for uploading data on arweave with no mints happening on the contract

Reply:
I am happy to take the hit for people rejecting transactions at the confirm in your own wallet stage. The payment, esp if we make the images small, is minimal.

Reply:
sounds good! I don't think it would cost alot though. if it turns out to be a problem we can always change this function